contracts/staking/NativeTokenStakingManager.sol

```diff
@@ -26,8 +26,13 @@ contract NativeTokenStakingManager is
         }
     }
 
+    /**
+     * @notice Initialize the ERC20 token staking manager
+     * @dev Uses reinitializer(2) on the PoS staking contracts to make sure after migration from PoA, the PoS contracts can reinitialize with its needed values.
+     * @param settings Initial settings for the PoS validator manager
+     */
     // solhint-disable ordering
-    function initialize(PoSValidatorManagerSettings calldata settings) external initializer {
+    function initialize(PoSValidatorManagerSettings calldata settings) external reinitializer(2) {
```

Comment:
Where does the version `2` actually get set when performing an upgrade? I don't see it in `ProxyAdmin.sol`, which is where I would expect to see it.

I'm trying to understand how `reinitializer` works under the hood.

Reply:
it's part of OZ's `Initializable` https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/utils/Initializable.sol#L152